### PR TITLE
Update gerrit to load a current cookiefile secret

### PR DESCRIPTION
**Pull request by fejta**
Also:
* Do the host=repo-a,repo-b parsing in projectsFlag
* Require all the required flags to be set
* Log whenever the token in the cookie changes
* Use a distroless base image
* Delete //prow/cmd/gerrit:push target (bump.sh already pushes this)

/assign @amwat @krzyzacy @BenTheElder

### CODE REVIEW

File: prow/cmd/gerrit/main.go [hunk 1]
```diff
@@ -34,52 +34,64 @@ import (
...
 
-	o.projects = make(map[string][]string)
-
-	for _, projects := range o.projectsVar {
-		split := strings.Split(projects, "=")
-		if len(split) != 2 {
-			return errors.New("--gerrit-projects must be in a form of --gerrit-projects=instance-foo=proj1,proj2")
-		}
+	if o.cookiefilePath == "" {
+		return errors.New("--cookiefile unset")
```
Comment (krzyzacy) [inline prow/cmd/gerrit/main.go]: I think empty is valid? (annoymous)?

Comment (fejta) [reply prow/cmd/gerrit/main.go]: Done

File: prow/cmd/gerrit/main.go [hunk 2]
```diff
@@ -34,67 +34,76 @@ import (
...
+	}
+	return strings.Join(hosts, " ")
 }
 
-func (p *projectsFlag) Set(value string) error {
-	*p = append(*p, value)
+func (p projectsFlag) Set(value string) error {
+	parts := strings.SplitN(value, "=", 2)
+	if len(parts) != 2 {
+		return fmt.Errorf("%s not in the form of host=repo-a,repo-b,...", value)
```
[the remark this answers is not in the document]
Comment (fejta) [reply prow/cmd/gerrit/main.go]: Done

**Pull request merged**